OONIProbeUnitTests/APITest.m

```diff
@@ -68,4 +69,39 @@ -(void)testDownloadUrls{
     }];
 }
 
+- (void)testDeleteJsons {
+    XCTestExpectation *expectation = [self expectationWithDescription:@"testDeleteJsons"];
+    int i = 4;
+    Measurement *existing_1 = [self addMeasurement:EXISTING_REPORT_ID];
+    Measurement *nonexisting_1 = [self addMeasurement:NONEXISTING_REPORT_ID];
+    Measurement *existing_2 = [self addMeasurement:EXISTING_REPORT_ID_2];
+    Measurement *nonexisting_2 = [self addMeasurement:NONEXISTING_REPORT_ID];
+    [TestUtility deleteUploadedJsons:^(Measurement *measurement){
+        if ([measurement.report_id isEqualToString:EXISTING_REPORT_ID] || [measurement.report_id isEqualToString:EXISTING_REPORT_ID_2])
+            XCTAssert(true);
+        else
+            XCTAssert(false);
+        i--;
+        if (i == 0)
+            [expectation fulfill];
+    } onError:^(NSError *error){
+        if ([measurement.report_id isEqualToString:NONEXISTING_REPORT_ID])
+            XCTAssert(true);
+        else
+            XCTAssert(false);
+        i--;
+        if (i == 0)
+            [expectation fulfill];
+    }];
```

Comment:
Let me know if my proposal below can help to test without making the code too hairy. Also, I believe that there is a singleton for measurements, so upon leaving this test you probably want to reset the database?

Reply:
This code was only a test, it would never worked. I will try your proposal and see if it works